saleor/graphql/product/mutations/products.py

```diff
@@ -1027,18 +1029,52 @@ def get_instance(cls, info, **data):
         """
 
         object_id = data.get("id")
-        if object_id and data.get("attributes"):
+        object_sku = data.get("sku")
+        if (object_id or object_sku) and data.get("attributes"):
             # Prefetches needed by AttributeAssignmentMixin and
             # associate_attribute_values_to_instance
             qs = cls.Meta.model.objects.prefetch_related(
                 "product__product_type__variant_attributes__values",
                 "product__product_type__attributevariant",
             )
-            return cls.get_node_or_error(
+
+            if object_id:
+                return cls.get_node_or_error(
+                    info, object_id, only_type="ProductVariant", qs=qs
+                )
+            if object_sku:
```

Comment:
This condition seems duplicated with what we have below, can't we improve this?

Reply:
Done